### PR TITLE
Sema: disallow @intFromPtr to accept comptime-only types

### PR DESCRIPTION
**Pull request by wrongnull**
It would make a lot more sense if we prevented it from accepting pointers to comptime-only types.

### CODE REVIEW

**Comment by wrongnull** [comment]
It looks like anytype parameters are treated as comptime-only under any conditions. @Vexu , is it a bug? If so, I could open an issue and try to fix one.

**Comment by Vexu** [comment]
No, this just has the same bug that you fixed for `@extern` in #17509. Also @SpexGuy was against this change when I previously tried to implement it in https://github.com/ziglang/zig/pull/11734#issuecomment-1139132094.

**Comment by mlugg** [comment]
> Also @SpexGuy was against this change when I previously tried to implement it in [#11734 (comment)](https://github.com/ziglang/zig/pull/11734#issuecomment-1139132094).

AFAICT that's about something else. `@intFromPtr` on pointers to comptime memory *where the child type is not comptime-only* could, in theory, be implemented in a semi-sane way using lazy values or something (although I still don't think that's a worthwhile idea). However, this is about `@intFromPtr` on an actual comptime-only type, where having a concrete address is just a nonsensical concept - it's not like you could ever perform meaningful arithmetic on that address, since the value doesn't have a size.

Currently, it seems backends (or at least LLVM) just lower these values as references to zero-bit types (so give them an arbitrary "fake address"), which is an entirely useless and unintuitive behavior.

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks! After a minor rewording of the error message, this is ready to land.

File: src/Sema.zig [hunk 1]
```diff
@@ -9891,6 +9891,17 @@ fn zirIntFromPtr(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!
     if (!ptr_ty.isPtrAtRuntime(mod)) {
         return sema.fail(block, ptr_src, "expected pointer, found '{}'", .{ptr_ty.fmt(mod)});
     }
+    const pointee_ty = ptr_ty.childType(mod);
+    if (try sema.typeRequiresComptime(ptr_ty)) {
+        const msg = msg: {
+            const msg = try sema.errMsg(block, ptr_src, "cannot accept pointer to comptime-only type '{}'", .{pointee_ty.fmt(mod)});
```
[the remark this answers is not in the document]
Comment (wrongnull) [reply src/Sema.zig:9897]: +

[the remark this answers is not in the document]
Comment (wrongnull) [reply src/Sema.zig:9897]: @andrewrk Is it worth mentioning this in the langref?

**Pull request merged**